pkg/model/config/config.go

```diff
@@ -48,6 +49,10 @@ type Config struct {
 	// This info is tracked only in project with version 2
 	Resources []GVK `json:"resources,omitempty"`
 
+	// CoreResourceVersions tracks the version of a core resource being used,
+	// ex. CustomResourceDefinition v1 vs. v1beta1.
+	CoreResourceVersions map[string][]string `json:"coreResourceVersions,omitempty"`
+
```

Comment:
IMO we cannot do that because it means breaking changes in the future. We do not need to have a specific attribute to store the core versions if we respect the model's domains as suggested in https://github.com/kubernetes-sigs/kubebuilder/pull/1644#discussion_r491227667.

- The default version used by the create api and create webhook plugins to scaffold the files ought to be defined in the plugin code, so that, we can have a default version for v2+ and another for v3+ and we will not need a migration guide step to describe to users update their PROJECT file when it changes.
- The design should allow we have a default version for each plugin(create api and create webhook) so that, we will not face problems to address scenarios such as the k8s api deprecate the v1 for webhooks and but not for CRDs.